aiida/manage/configuration/__init__.py

```diff
@@ -59,6 +59,27 @@
 BACKEND_UUID = None  # This will be set to the UUID of the profile as soon as its corresponding backend is loaded
 
 
+def check_version():
+    """Check the currently installed version of ``aiida-core`` and warn if it is a post release development version.
+
+    The ``aiida-core`` package maintains the protocol that the ``develop`` branch will use a post release version
+    number. This means it will always append `.post0` to the version of the latest release. This should mean that if
+    this protocol is maintained properly, this method will print a warning if the currently installed version is a
+    post release development branch and not an actual release.
+    """
+    from packaging.version import parse
+
+    from aiida import __version__
+    from aiida.cmdline.utils import echo
+
+    version = parse(__version__)
+
+    if version.is_postrelease:
+        echo.echo_warning(f'You are currently using a post release development version of AiiDA: {version}')
+        echo.echo_warning('Be aware that this is not recommended for production and is not officially supported.')
+        echo.echo_warning('Databases used with this version may not be compatible with future releases of AiiDA.\n')
```

Comment:
Maybe say explicitly 'and you might not be able to migrate your data to a production version of AiiDA'? Or something like this?